### PR TITLE
Pin Numpy to 1.20.* for wheel testing

### PR DESCRIPTION
**Pull request by astrofrog**
### Description

Alternative to https://github.com/astropy/astropy/pull/12718 - this instead pins Numpy to a version known to work on both 32- and 64-bit platforms. We can remove this pin once we drop support for 32-bit wheels.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by astrofrog** [comment]
**Note: do not merge until the TMP commit is removed!**

**Comment by astrofrog** [comment]
Looks like this worked so I will remove the temporary commit and mark this as ready for review shortly

File: azure-pipelines.yml [hunk 1]
```diff
@@ -41,6 +41,9 @@ variables:
   CIBW_MANYLINUX_I686_IMAGE: manylinux2010
   CIBW_ARCHS_LINUX: "auto, aarch64"
   CIBW_ARCHS_MACOS: "x86_64 arm64"
+  # Numpy 1.22 doesn't have wheels for i386, so we pin Numpy to an older version
+  # that does - this pin can be removed once we drop support for 32-bit wheels.
+  CIBW_TEST_REQUIRES: "numpy==1.20.*"
```
Comment (pllim) [inline azure-pipelines.yml:46]: This is only for *testing* (not building) the wheels, right?

**Comment by mhvk** [comment]
This seems a good approach. Longer term it may easiest to stop building for `i386` once our minimum version of numpy no longer supports it.

File: azure-pipelines.yml (hunk 1 above)
Comment (saimn) [reply azure-pipelines.yml:46]: Yes, this is for tests run in the wheel building container.

**Review by saimn — APPROVED** [review]
Sounds good to me.

**Review by pllim — APPROVED** [review]

**Pull request merged**